notification/src/handlers/amqpHandler.ts

```diff
@@ -0,0 +1,149 @@
+import { StorageHandler } from "./storageHandler"
+import { Channel, connect, Connection } from "amqplib/callback_api"
+import { TransformationEvent } from '../interfaces/transformationResults/transformationEvent';
+import JSNotificationService from '../jsNotificationService';
+import VM2SandboxExecutor from "../vm2SandboxExecutor";
+import { CONFIG_TYPE } from "../models/notificationConfig";
+
+/**
+ * This class handles the communication with the AMQP service (rabbitmq)
+ * It interacts with this channels:
+ *
+ *      * Notification Channel:
+ *       ----------------------
+ *       A channel to notify the notification service that a transformation is done.
+ *       (see TransformationEvent for details of the Event).
+ *      
+ */
+export class AmqpHandler{
+    notifQueueName = process.env.AMQP_NOTIFICATION_QUEUE!     // Queue name of the Job Queue
+
+    /**
+     * Connects to Amqp Service and initializes a channel
+     * 
+     * @param retries   Number of retries to connect to the notification-config db
+     * @param backoff   Time to wait until the next retry
+     */
+    public async connect(retries: number, backoff: number) {
+        const rabbit_url = process.env.AMQP_SERVICE_HOST;
+        const rabbit_usr = process.env.AMQP_SERVICE_USER;
+        const rabbit_password = process.env.AMQP_SERVICE_PWD;
+        const rabit_amqp_url = 'amqp://' + rabbit_usr + ':' + rabbit_password + '@' + rabbit_url;
+
+        console.log("URL: " + rabit_amqp_url)
+
+        var established: boolean = false    // amqp service connection result
+        const handler: AmqpHandler = this   // for ability to access methods and members in callback
+        
+        for (let i = 0; i < retries; i++) {
+            await this.backOff(backoff)
+            await connect(rabit_amqp_url, function (error0: any, connection: Connection) {
```

Comment:
if you replace `function (error0, connection) { ... }` with `(error0, connection) => { ... }` you can still access the class members and functions (here AmqpHandler) with `this` and don't need the self-reference as variable beforehand ;)